src/app/plugins/shadowban.plugin.ts

```diff
@@ -64,20 +71,25 @@ export class ShadowBanPlugin extends Plugin {
       return this.BANNED_CATEGORIES.some((n) => chanName === n);
     });
 
-    catsToBan
-      .reduce(function(acc: GuildChannel[], cat: CategoryChannel) {
-        return [...acc, ...cat.children.array()];
-      }, [])
-      .forEach((chan) => callback(chan, user));
+    const promises = catsToBan.reduce((acc, cat: CategoryChannel) => {
+      acc.push(...cat.children.array().map((chan: GuildChannel) => callback(chan)));
+      return acc;
+    }, []);
+
+    await Promise.all(promises);
   }
 
-  private _banUser(chan: GuildChannel, user: User) {
-    chan.overwritePermissions(user, {
-      VIEW_CHANNEL: false,
-    });
+  private _banUser(user: User) {
+    return (chan: GuildChannel) => {
+      chan.overwritePermissions(user, {
```

Comment:
`overwritePermissions` returns a promise. You should await it.

```
return async (chan) => {
  await chan.overwritePermissions(...);
}
```

https://discord.js.org/#/docs/main/stable/class/GuildChannel?scrollTo=overwritePermissions

Reply:
Still slightly figuring out promises lol, thought that `await Promise.all` was enough